File: C2009G/adf1/29-05-2021/src/main/java/models/Point.java
package models;

public class Point {
    private Double x, y;
    //ten phuong thuc khoi tao(constructor)
    public Point(Double x, Double y) {
        this.x = x;
        this.y = y;
    }

    public Double getX() {
        return x;
    }

    public void setX(Double x) {
        this.x = x;
    }

    public Double getY() {
        return y;
    }

    public void setY(Double y) {
        this.y = y;
    }
    public Double getDistance(Point p) {
        return Math.sqrt(Math.pow(this.x - p.getX(), 2) + Math.pow(this.y - p.getY(), 2));
    }
    @Override
    public String toString() {
        return String.format("x = %f, y = %f", x, y);
    }
}

File: C2108G2/AP-C#/02-02-2023/ConsoleApp/Models/Point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp.Models
{
    internal class Point
    {
        private int x, y;
        //constructor
        /*
        public Point(int x, int y) {
            //java's style            
            this.x = x;
            this.y = y;
        }
        */
        //getter, setter in C#
        /*
        public int X {
            get {
                return x;
            }
            set { 
                x = value;
            }
        }
        */
        //getter, setter in C#(simplified)
        public int X { get; set; }
        public int Y
        {
            get
            {
                return y;
            }
            set
            {
                y = value;
            }
        }
        /*
        public int getX() {
            //DON't DO THIS IN C# !            
            return x;
        }
        public int getY()
        {
            return y;
        }
        public void setX(int x)
        {
            this.x = x;
        }
        public void setY(int y)
        {
            this.y = y;
        }
        */
       
        public override string ToString()
        {
            return  $"x = {X},"+
                    $"y = {Y}"
            ;
        }
    }
}
